card-game-server/src/main/java/com/cardgameserver/dao/GameRoomDao.java

```diff
@@ -0,0 +1,67 @@
+package com.cardgameserver.dao;
+
+import com.cardgameserver.model.GameRoom;
+import com.cardgameserver.util.ConnectionManager;
+
+import java.sql.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GameRoomDao {
```

Comment:
🔥 We need to implement IRepositoy interface for every Dao.

I think this will make you edit few method name (by overriding method)

```
[1 mark] Introduce MVC, Factory, DAO Patterns in the backend code using the following link as a reference:

https://dzone.com/articles/building-simple-data-access-layer-using-jdbc

Hint:  You will qualify for full marks, if the following are incorporated.

The servlet responsible for managing quizzes shall not refer to JDBC API at all i.e. it should not have "import java.sql.*".
JDBC code shall be abstracted in the Repository class.  Repository class shall implement an IRepository interface that declares methods such as init(string connectString), close(), insert(), update(), delete() and select() etc.  The init() method should take care of getting the JDBC connection object.  The close() method shall close the database connection.
[REVISED] There should be a Quiz class which should inherit from an abstract class.  Lets call this class Game (or AClass).  This abstract class shall declare an abstract method "public abstract String serialize()".  The Quiz class shall implement this method by returning a (preferably Json) string containing the state of the object.  The parameters of the Repository class then could be the reference to this abstract class.  GIVEN THE TIME CONSTRAINTS AND THE STAT HOLIDAYS, THIS PART IS OPTIONAL BUT MAKE SURE YOU REVIEW the DaoPattern.java attached to the week 3 folder TO STUDY THE DESIGN AS THERE MAY BE A MIDTERM QUESTION ON THIS.  
```